field/composite.go

```diff
@@ -515,7 +516,28 @@ func (f *Composite) packByTag() ([]byte, error) {
 	return packed, nil
 }
 
-func (f *Composite) unpack(data []byte, isVariableLength bool) (int, error) {
+// wrapErrorUnpack calls the core unpacking logic and wraps any
+// errors in a *UnpackError. It assumes that the mutex is already
+// locked by the caller.
+func (f *Composite) wrapErrorUnpack(src []byte, isVariableLength bool) (int, error) {
+	offset, tagID, err := f.unpack(src, isVariableLength)
+	if err != nil {
+		subfields := []string{}
+		var unpackErr *mooverrors.UnpackError
+		if errors.As(err, &unpackErr) {
+			subfields = unpackErr.FieldIDs
+		}
+		return offset, &mooverrors.UnpackError{
+			Err:        err,
+			FieldID:    tagID,
+			FieldIDs:   append([]string{tagID}, subfields...),
+			RawMessage: src,
+		}
```

Comment:
```suggestion
		return offset, &mooverrors.UnpackError{
			Err:        err,
			FieldID:    tagID,
			RawMessage: src,
		}
```

please, read the main comment that will explain this suggestion